File: 1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for i in xrange(len(nums)):
            dic[nums[i]] = i
        for i in xrange(len(nums)):
            if target - nums[i] in dic and i < dic[target - nums[i]]:
                return [i, dic[target - nums[i]]]

File: 1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> dic;
        for (int i = 0; i < nums.size(); i++)
            dic[nums[i]] = i;
        for (int i = 0; i < nums.size(); i++)
        {
            if (dic.count(target - nums[i]) > 0 && i < dic[target - nums[i]])
                return {i, dic[target - nums[i]]};
        }
        return {};
    }
};
